src/Command/ErrorFormatter/JunitErrorFormatter.php

```diff
@@ -0,0 +1,88 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command\ErrorFormatter;
+
+use PHPStan\Command\AnalysisResult;
+use PHPStan\Command\Output;
+use PHPStan\File\RelativePathHelper;
+use function sprintf;
+
+class JunitErrorFormatter implements ErrorFormatter
+{
+
+	/** @var \PHPStan\File\RelativePathHelper */
+	private $relativePathHelper;
+
+	public function __construct(RelativePathHelper $relativePathHelper)
+	{
+		$this->relativePathHelper = $relativePathHelper;
+	}
+
+	public function formatErrors(
+		AnalysisResult $analysisResult,
+		Output $output
+	): int
+	{
+		$result = '<?xml version="1.0" encoding="UTF-8"?>';
+		$result .= sprintf(
+			'<testsuite failures="%d" name="phpstan" tests="%d" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/junit-team/junit5/r5.5.1/platform-tests/src/test/resources/jenkins-junit.xsd">',
+			$analysisResult->getTotalErrorsCount(),
+			$analysisResult->getTotalErrorsCount()
+		);
+
+		foreach ($analysisResult->getFileSpecificErrors() as $error) {
+			$fileName = $this->relativePathHelper->getRelativePath($error->getFile());
+			$result .= $this->createTestCase(
+				sprintf('%s:%s', $fileName, (string) $error->getLine()),
+				$this->escape($error->getMessage())
+			);
+		}
+
+		foreach ($analysisResult->getNotFileSpecificErrors() as $genericError) {
+			$result .= $this->createTestCase('Generic error', $this->escape($genericError));
```

Comment:
Rather `general`, `generic` already means something in the typesystem :)

Reply:
FYI I complete the change, but I see other formatter use *generic* instead of *general*, so I prefer to keep consistency here than migrate generic to general in all formatter, WDYT?

Reply:
Where is `generic` used in the context of error formatters?

Reply:
My fault, is used only on text reported by default error generator, sorry!